packages/components/src/modal/use-modal-exit-animation.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * WordPress dependencies
+ */
+import { useReducedMotion } from '@wordpress/compose';
+import { useCallback, useRef, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { CONFIG } from '../utils';
+
+// Animation duration (ms) extracted to JS in order to be used on a setTimeout.
+const FRAME_ANIMATION_DURATION = CONFIG.transitionDuration;
+const FRAME_ANIMATION_DURATION_NUMBER = Number.parseInt(
+	CONFIG.transitionDuration
+);
+
+const EXIT_ANIMATION_NAME = 'components-modal__disappear-animation';
+
+export function useModalExitAnimation() {
+	const frameRef = useRef< HTMLDivElement >();
+	const [ isAnimatingOut, setIsAnimatingOut ] = useState( false );
+	const isReducedMotion = useReducedMotion();
+
+	const closeModal = useCallback(
+		( onAnimationEnd: () => void ) => {
+			function handleAnimationEnd( e: AnimationEvent ) {
+				if ( e.animationName === EXIT_ANIMATION_NAME ) {
+					setIsAnimatingOut( false );
+					onAnimationEnd();
+				}
+			}
+
+			if ( isReducedMotion ) {
+				onAnimationEnd();
+				return;
+			}
+
+			if ( ! frameRef.current ) {
+				return;
+			}
+
+			// Grab a reference to the frame element, to make sure we're referencing
+			// the same reference in the cleanup function.
+			const frameEl = frameRef.current;
+
+			setIsAnimatingOut( true );
+			frameEl.addEventListener( 'animationend', handleAnimationEnd, {
+				once: true,
+			} );
+			const animationTimeout = window.setTimeout( () => {
+				setIsAnimatingOut( false );
+				onAnimationEnd();
+			}, FRAME_ANIMATION_DURATION_NUMBER );
+
+			return () => {
+				frameEl.removeEventListener(
+					'animationend',
+					handleAnimationEnd
+				);
+				window.clearTimeout( animationTimeout );
+			};
```

Comment:
I don't see this cleanup called anywhere. The `closeModal` function is used as an event handler, not as an effect handler.

The cleanup should happen in some `useEffect` inside `useModalExitAnimation`.

Reply:
Absolutely — that was a remnant of a refactor from `useEffect` that I forgot to clean up, thank you for spotting it.

I believe this comment should be resolved by the code changes in https://github.com/WordPress/gutenberg/pull/65203/commits/0ab921dadfde8e40c7a69714b1ded43d88f2c701